mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1707,15 +1708,19 @@ def _evaluate(
                 elif self.model_type == _ModelType.TEXT:
                     self.builtin_metrics = text_metrics
                 elif self.model_type == _ModelType.RETRIEVER:
-                    k = self.evaluator_config.pop("k", 3)  # default k to 3 if not specified
-                    if not (isinstance(k, int) and k > 0):
+                    # default k to 3 if not specified
+                    retriever_k = self.evaluator_config.pop("retriever_k", 3)
+                    if not (isinstance(retriever_k, int) and retriever_k > 0):
                         _logger.warning(
-                            "Cannot calculate 'precision_at_k' for invalid parameter 'k'."
-                            f"'k' should be a positive integer; found: {k}"
-                            "Skipping metric logging."
+                            "Cannot calculate 'precision_at_k' and 'recall_at_k' for invalid "
+                            f"parameter 'retriever_k'. 'retriever_k' should be a positive integer; "
+                            f"found: {retriever_k}. Skipping metric logging."
```

Comment:
This validation helps for `model_type="retriever", but I tested with `extra_metrics` and can still specify k=0, k=-2, etc. For example:

```
import pandas as pd
import mlflow

X = pd.DataFrame({"question": ["q1?"] * 3, "ground_truth": [["doc1", "doc2"]] * 3})

def fn(X):
    return {"retrieved_context": [["doc1", "doc3", "doc2"]] * len(X)}

with mlflow.start_run() as run:
    results = mlflow.evaluate(
        model=fn,
        data=X,
        targets="ground_truth",
        extra_metrics=[mlflow.metrics.precision_at_k(-1), mlflow.metrics.recall_at_k(-2)],
    )
run = mlflow.get_run(run.info.run_id)
print(run.data.metrics)
```

yields

```
{'precision_at_-1/variance': 0.0, 'precision_at_-1/mean': 0.5, 'precision_at_-1/p90': 0.5, 'recall_at_-2/variance': 0.0, 'recall_at_-2/mean': 0.5, 'recall_at_-2/p90': 0.5}

```

Reply:
Can we add validation to ensure that constructing `mlflow.metrics.precision_at_k(K<=0 || k non-integer)` throws?

Reply:
yep!